control/server.py

```diff
@@ -243,3 +246,48 @@ def _ping(self):
         except Exception as ex:
             self.logger.error(f"spdk_get_version failed with: \n {ex}")
             return False
+
+    def gateway_rpc_caller(self, requests, is_add_req):
+        """Passes RPC requests to gateway service."""
+        for key, val in requests.items():
+            if key.startswith(GatewayState.BDEV_PREFIX):
+                if is_add_req:
+                    req = json_format.Parse(val, pb2.create_bdev_req())
+                    self.gateway_rpc.create_bdev(req)
+                else:
+                    req = json_format.Parse(val,
+                                            pb2.delete_bdev_req(),
+                                            ignore_unknown_fields=True)
```

Comment:
The original `create_X_req` / `add_X_req` is used to generate the respective delete/remove request when an update identifies its removal from the OMAP. Since the create/add requests for bdev, subsystem, and namespace have more fields than their delete/remove counterparts, those can be ignored when building the delete requests.